File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// common_len은 배열 common의 길이입니다.
int solution(int common[], size_t common_len) {
    int answer = 0;
    
    if(common[2] - common[1] == common[1] - common[0]){
        answer = common[common_len - 1] + (common[1] - common[0]);
    }else {
        answer = common[common_len - 1] / common[1] * common[2];
    }
    
    return answer;
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    answer = 0
    
    if common[2] - common[1] == common[1] - common[0]:
        return common[-1] + common[1] - common[0]
    elif common[2] / common[1] == common[1] / common[0]:
        return common[-1] * common[1] / common[0]
    
    return answer